### PR TITLE
feat(core): busy indicator horizon

### LINKED ISSUE #9827: Component Design Update: Busy Indicator
- [ ] review markup changes from fd-styles
- [ ] Horizon specs(visual and interaction) updates
- [ ] Delta theming
- [ ] identify and document breaking changes

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

closes #9827
part of #9823 

## Description

Migrated some parts to standalone and updated busy indicator. added example to the dialog module for added example on fd-styles side

### CODE REVIEW

File: libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts [hunk 1]
```diff
@@ -20,35 +30,82 @@ import { coerceElement } from '@angular/cdk/coercion';
...
+            .pipe(
+                map((content) => {
+                    portalOutlet.detach();
+                    if (content) {
+                        if (typeof content === 'string') {
+                            const fdTextElement = this.renderer.createElement('p');
+                            fdTextElement.classList.add('fd-text');
+                            fdTextElement.innerHTML = content;
+                            this.renderer.appendChild(this.elementRef.nativeElement, fdTextElement);
+                            return new DomPortal(fdTextElement);
```
Comment (N1XUS) [inline libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts:95]: Not sure if this should be here. Logic looks out of scope of this component

File: libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts [hunk 2]
```diff
@@ -20,35 +30,82 @@ import { coerceElement } from '@angular/cdk/coercion';
...
+                            return new DomPortal(coerceElement(content as HTMLElement | ElementRef<HTMLElement>));
+                        } else if (content instanceof TemplateRef) {
+                            return new TemplatePortal(content, this.viewContainerRef);
+                        }
+                        return new ComponentPortal(content);
+                    }
+                    return null;
+                }),
+                filter(Boolean),
+                tap((portal) => portalOutlet.attach(portal)),
```
Comment (N1XUS) [inline libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts:106]: Using `attach` in `tap` looks less clear to me than in `subscribe` callback

File: libs/core/src/lib/dialog/utils/dialog-ref.class.ts [hunk 3]
```diff
@@ -37,9 +38,22 @@ export class DialogRef<T = any, P = any> extends DialogRefBase<T, P> {
...
      * Displays the dialog in loading state.
-     * @param isLoading Value used to determine if dialog window should be displayed in loading state.
+     * @param loadingData Value used to determine if dialog window should be displayed in loading state.
      */
-    loading(isLoading: boolean): void {
-        this._onLoading.next(isLoading);
+    loading(
+        loadingData:
+            | boolean
+            | {
```
Comment (N1XUS) [inline libs/core/src/lib/dialog/utils/dialog-ref.class.ts:46]: Maybe move to a type or interface?

File: libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts (hunk 2 above)
Comment (g-cheishvili) [reply libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts:106]: Sure, I just prefer tap for side things

File: libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts (hunk 1 above)
Comment (g-cheishvili) [reply libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts:95]: Yes, you are totally right

**Comment by InnaAtanasova** [comment]
The PR link to the docs is not working, can't test the component

**Review by InnaAtanasova — APPROVED** [review]

**Comment by g-cheishvili** [comment]
I can not make e2e fail locally, even with the same exact target
